File: day2/part1.py
def smallestArea(l: int, w: int, h: int) -> int:
    a1 = l*w
    a2 = w*h
    a3 = h*l

    if (a1 < a2 and a1 < a3):
        return a1
    elif (a2 < a1 and a2 < a3):
        return a2
    elif (a3 < a2 and a3 < a1):
        return a3


def main():
    with open('input.txt', 'r') as f:
        p = f.read().split('\n')

    totalArea = 0
    for item in p:
        l, w, h = map(int, item.split('x'))
        totalArea += 2*l*w + 2*w*h + 2*l*h + smallestArea(l, w, h)

    print(totalArea)


main()

File: day2/part1.go
package main

import (
	"fmt"
	"strconv"
	"strings"

	"github.com/sagnikc395/aoc2015.git/utils"
)

func smallestSideArea(l, b, h int) int {
	return min(l*b, b*h, h*l)
}

func surfaceArea(l, b, h int) int {
	return 2*l*b + 2*b*h + 2*h*l
}

func main() {
	f := utils.ReadFile("./input.txt")

	data := strings.Split(string(f), "\n")

	totalCount := 0
	for _, item := range data {
		dims := strings.Split(item, "x")
		l, _ := strconv.Atoi(dims[0])
		b, _ := strconv.Atoi(dims[1])
		h, _ := strconv.Atoi(dims[2])
		surface_area := surfaceArea(l, b, h)
		smallestSide_area := smallestSideArea(l, b, h)
		totalCount += surface_area + smallestSide_area
	}
	fmt.Println(totalCount)
}
